### DISCUSSION TITLE
How to add citation within code chunk which is flextable()?

### DISCUSSION THREAD

**Discussion by kellygarvey** [quarto-dev/quarto-cli#6127] [Q&A]
### Description

I am wondering if a citation [@____] can be added to a code chunk creating a flextable(). I would like the citation to be in the footer. I can add a manual quote with add_footer_lines but I want the citation to be linked. 

```{r Missing data table}
#| tbl-cap: Summary of missing values in Montana data
#| label: tbl-missingness-all-col
montana %>%
        sapply(function(x) sum(is.na(x))) %>%
        data.frame(Column = colnames(montana),
                   Missing_Sum = .,
                   Missing_Percent = ./nrow(montana) * 100) %>%
        arrange(desc(Missing_Sum)) %>%
        mutate(Missing_Sum = round(Missing_Sum, 2),
               Missing_Percent = round(Missing_Percent, 2)) %>%
        flextable() %>%
        set_header_labels(Column = "Column",
                          Missing_Sum = "# NA values",
                          Missing_Percent = "% NA values")%>%
        color(i = c(14, 15, 20, 21, 28), j = 1:3, color = "red")%>%
        add_footer_lines("Red indicates exclusion based fields")%>%
        autofit()
```

**Comment by cderv** [comment]
@rich-iannone @cscheid is the new mechanism to pass information from Tables to Quarto already available ? I know you worked on something for **gt** and I wonder if this is ready to help bring support to this in **flextable**. 

Thanks !

**Reply by cscheid** [reply]
If flextable emits html tables in a format that pandoc can parse, then yes, that should be available.

**Reply by cderv** [reply]
Oh ok ! I thought that would require the _messaging_ feature your worked on for **gt**. I'll look into what **flextable** is doing then

But if Quarto parse the HTML table - I guess this would mean that all the styling and customization that **flextable** brings will be lost... 🤔

**Reply by mcanouil** [reply]
You probably need to pass the `data-quarto-disable-processing="true"` attribute to it.

**Reply by cscheid** [reply]
> Oh ok ! I thought that would require the messaging feature your worked on for gt. I'll look into what flextable is doing then

That part isn't ready yet (that's the reason I qualified my answer with "If flextable emits html tables in a format that pandoc can parse"). We still need to implement the feature where libraries can communicate out-of-band (for example, in the case of interactive tables, where the output will just be a pile of javascript)
